### ISSUE TITLE
ZeroDivisionError in NotebookProgressBar

### ISSUE THREAD

**Issue by pete88b** [huggingface/transformers#27513]
### System Info

- `transformers` version: 4.35.1
- Platform: Windows-10-10.0.19041-SP0
- Python version: 3.9.0
- Huggingface_hub version: 0.17.3
- Safetensors version: 0.3.1
- Accelerate version: 0.20.3
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes (GeForce 3080 laptop)
- Using distributed or parallel set-up in script?: no

### Who can help?

Hi @muellerzr - hope this is an easy one for you (o:

I'm sometimes seeing a divide by zero error when NotebookProgressBar updates.

I've suggested changes to `transformers/utils/notebook.py` in [my notebook](https://github.com/pete88b/spelling-mistake-maker/blob/main/tmp/NotebookProgressBar_divide_by_zero_issue.ipynb).

I'm guessing that for a change this simple, it'll be easier for you to make the change - but please let me know if you prefer I make a PR for this change

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Please see [my notebook](https://github.com/pete88b/spelling-mistake-maker/blob/main/tmp/NotebookProgressBar_divide_by_zero_issue.ipynb)

### Expected behavior

We should not get a divide by zero error when `average_time_per_item` is zero

**Comment by amyeroberts** [comment]
Hi @pete88b please feel free to open a PR and ping me for review!

**Comment by hi-sushanta** [comment]
Hi Peter, I have reviewed your code and executed it successfully on my end. Please take a look at my notebook and check whether it works as expected on your side.
[notebook](https://colab.research.google.com/drive/1x7fG6GeLDSKXWl6KgsBArIb5qh5OEMdQ?usp=sharing)

**Comment by pete88b** [comment]
Hi @hi-sushanta #27524 fixes `update` but misses the fix to `self.label += f", {1/self.average_time_per_item:.2f} it/s"` in `update_bar` - there's more detail in the linked notebook in case it helps.
if it's not too late, could you update your PR to include both (o:

**Comment by hi-sushanta** [comment]
Thank you, Peter🤗.
I updated my code similar to you did.
